jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/EthereumAuthenticator.java

```diff
@@ -0,0 +1,771 @@
+    }
+
+    public void setAuthenticationPath(String authenticationPath)
+    {
+        if (authenticationPath == null)
+        {
+            authenticationPath = _authenticationPath;
+            LOG.warn("authentication path must not be null, defaulting to {}", authenticationPath);
+        }
+        else if (!authenticationPath.startsWith("/"))
+        {
+            authenticationPath = "/" + authenticationPath;
+            LOG.warn("authentication path must start with /");
+        }
+
+        _authenticationPath = authenticationPath;
+    }
+
+    public void setNoncePath(String noncePath)
+    {
+        if (noncePath == null)
+        {
+            noncePath = _noncePath;
+            LOG.warn("nonce path must not be null, defaulting to {}", noncePath);
+        }
+        else if (!noncePath.startsWith("/"))
+        {
+            noncePath = "/" + noncePath;
+            LOG.warn("nonce path must start with /");
+        }
+
+        _noncePath = noncePath;
+    }
+
+    public void setMaxMessageSize(int maxMessageSize)
+    {
+        _maxMessageSize = maxMessageSize;
+    }
+
+    public void setDispatch(boolean dispatch)
+    {
+        _dispatch = dispatch;
+    }
+
+    public void setLogoutRedirectPath(String logoutRedirectPath)
+    {
+        if (logoutRedirectPath != null && !logoutRedirectPath.startsWith("/"))
+        {
+            LOG.warn("logout redirect path must start with /");
+            logoutRedirectPath = "/" + logoutRedirectPath;
+        }
+
+        _logoutRedirectPath = logoutRedirectPath;
+    }
+
+    public void setErrorPage(String path)
+    {
+        if (path == null || path.trim().isEmpty())
+        {
+            _errorPath = null;
+            _errorPage = null;
+        }
+        else
+        {
+            if (!path.startsWith("/"))
+            {
+                LOG.warn("error-page must start with /");
+                path = "/" + path;
+            }
+            _errorPage = path;
+            _errorPath = path;
+            _errorQuery = "";
+
+            int queryIndex = _errorPath.indexOf('?');
+            if (queryIndex > 0)
+            {
+                _errorPath = _errorPage.substring(0, queryIndex);
+                _errorQuery = _errorPage.substring(queryIndex + 1);
+            }
+        }
+    }
+
+    @Override
+    public UserIdentity login(String username, Object credentials, Request request, Response response)
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug("login {} {} {}", username, credentials, request);
+
+        UserIdentity user = super.login(username, credentials, request, response);
+        if (user != null)
+        {
+            Session session = request.getSession(true);
+            AuthenticationState cached = new SessionAuthentication(getAuthenticationType(), user, credentials);
+            synchronized (session)
+            {
+                session.setAttribute(SessionAuthentication.AUTHENTICATED_ATTRIBUTE, cached);
```

Comment:
We will probably receive the "standard" security report there that a malicious user can create a new session with huge credentials and then never complete authentication.  We will then collect sessions with large content.

Can we add size limits on the credentials?  Also I think we have a mechanism to have a shorter session idle timeout for unauthenticated sessions.  Make sure that is activate and/or mentioned in the documentation.

Reply:
The credentials consist of an ethereum address which is exactly 40 characters long, enforced by the parser.
And this is specified in the spec.

Besides this particular code only happens when the authentication is complete.

But if you look at the nonce path, there is a `Set<String>` (limited at max length of 5), which is stored in the session to remember the nonces. Which is probably more of concern than this.